src/build/__init__.py

```diff
@@ -13,23 +13,29 @@
 import difflib
 import logging
 import os
-import re
 import subprocess
 import sys
-import textwrap
-import types
 import warnings
 import zipfile
 
-from collections import OrderedDict
-from collections.abc import Iterator, Set
-from typing import Any, Callable, Mapping, MutableMapping, Optional, Sequence, Tuple, Type, Union
+from collections.abc import Iterator
+from typing import Any, Callable, Mapping, Optional, Sequence, TypeVar, Union
 
 import pyproject_hooks
 
+from . import env
+from ._exceptions import (
+    BuildBackendException,
+    BuildException,
+    BuildSystemTableValidationError,
+    FailedProcessError,
+    TypoWarning,
+)
+from ._util import check_dependency, parse_wheel_filename
```

Comment:
We already have a `util`, also having a `_util` seems confusing? If it's important to hide these, they could be `_check_dependency` and `_parse_wheel_filename` in `util`? Or maybe they could be in a file with a better private name?

Having public and private utils separated in this way isn't terrible though, just checking.

Reply:
Yes they can, we just need `__all__` (which we should be adding anyway, along with `__dir__`).

Reply:
But that would make them public assuming they are contained in a public (i.e. ununderscored) module.

Reply:
I'd say that an underscored name in a `__all__` is still hidden, it just means it is used from other modules. Using it downstream is just as risky as if there was no `__all__`. `__all__` describes what is used by other modules - things not in __all__ should only be used in the file they reside in.

You could even filter things that start with an underscore in `__dir__`, but I'd not bother, things like IDEs and IPython still filter them by default.

Reply:
Are they also hidden from Sphinx?

Reply:
I think having a private module for internal use is cleaner FWIW, but I don't feel strongly about this.

Reply:
I'm fine with it either way. Just a note.

Sphinx hides things starting with an underscore by default, but it's configurable, IIRC.